### DISCUSSION TITLE
Navigating to different route is interfering with the overlay click event.

### DISCUSSION THREAD

**Discussion by gagan-smarter-codes** [gilbarbara/react-joyride#903] [Q&A] [answered]
I'm using react-joyride with following callback functions:-

```
const handleCallback = (data) => {
    const { action, index, lifecycle, type } = data;
    if (action === "reset" && type == "tour:status") {
      //this event is fired when a user clicks on the overlay to close the onboarding or the onboarding is reset.
      setSteps([]);
      setRun(false);
    }

     if (action === "close" && type === "error:target_not_found" && index == 1) {
        //second step in onboarding is not found.
        navigate("settings/channels");
     }
}
```
I noticed that `error:target_not_found` is getting fired when I'm clicking on overlay even though it shouldn't happen as I'm not trying to access second step of the onboarding process.
How can I ensure that both of them do not interfere with each other?

**Comment by gilbarbara** [comment, marked as answer]
Hey @gagan-smarter-codes 

By default clicking the overlay will trigger a close event.
Have you checked the [docs](https://docs.react-joyride.com/props)?
There's a `disableOverlayClose` prop that you can set globally in the component or individually for each step.
